codersdk/toolsdk/chatgpt.go

```diff
@@ -0,0 +1,340 @@
+package toolsdk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/google/uuid"
+
+	"github.com/coder/aisdk-go"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+func getServerURL(deps Deps) string {
+	serverURLCopy := *deps.coderClient.URL
+	serverURLCopy.Path = ""
+	serverURLCopy.RawQuery = ""
+	return serverURLCopy.String()
+}
```

Comment:
We may want to add a new function to `Deps`.
Then we could just `deps.ServerURL()` to get it.